File: emp_front/lib/model/employee.dart
class Employe {
  final int id;
  final String nom;
  final String prenom;
  final String sex;
  final int age;
  final int tel;
  final String mail;
  final String fonction;

  Employe(
      {required this.nom,
      required this.prenom,
      required this.sex,
      required this.age,
      required this.tel,
      required this.mail,
      required this.fonction,
      required this.id});

  Employe.fromJson(Map<String, dynamic> json)
      : this(
          nom: json['nom'].toString(),
          prenom: json['prenom'].toString(),
          sex: json['sex'].toString(),
          age: json['age'],
          tel: json['tel'],
          mail: json['mail'].toString(),
          fonction: json['fonction'].toString(),
          id: json['id'],
        );

  Map<String, dynamic> toJson() {
    return {
      'nom': nom,
      'prenom': prenom,
      'sex': sex,
      'age': age.toString(),
      'tel': tel.toString(),
      'mail': mail,
      'fonction': fonction,
    };
  }
}

File: emp-back/src/main/java/com/lundu/empback/entities/Employee.java
package com.lundu.empback.entities;

import com.lundu.empback.dto.request.EmployeeRequestDTO;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "employee")
@AllArgsConstructor
@NoArgsConstructor
@Data
public class Employee {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String nom;
    private String prenom;
    private String sex;
    private int age;
    private int tel;
    private String mail;
    private String fonction;

    public Employee(String nom, String prenom, String sex, int age, int tel, String mail, String fonction) {
        this.nom = nom;
        this.prenom = prenom;
        this.sex = sex;
        this.age = age;
        this.tel = tel;
        this.mail = mail;
        this.fonction = fonction;
    }

    public static Employee fromDTO(EmployeeRequestDTO employeeRequestDTO) {
        Employee employee = new Employee();
        employee.setNom(employeeRequestDTO.nom());
        employee.setPrenom(employeeRequestDTO.prenom());
        employee.setSex(employeeRequestDTO.sex());
        employee.setAge(employeeRequestDTO.age());
        employee.setTel(employeeRequestDTO.tel());
        employee.setMail(employeeRequestDTO.mail());
        employee.setFonction(employeeRequestDTO.fonction());
        return employee;
    }

    public void update(EmployeeRequestDTO employeeRequestDTO) {
        if(employeeRequestDTO.nom() != null) this.nom = employeeRequestDTO.nom();
        if(employeeRequestDTO.prenom() != null) this.prenom = employeeRequestDTO.prenom();
        if(employeeRequestDTO.sex() != null) this.sex = employeeRequestDTO.sex();
        if(employeeRequestDTO.age() != 0) this.age = employeeRequestDTO.age();
        if(employeeRequestDTO.tel() != 0) this.tel = employeeRequestDTO.tel();
        if(employeeRequestDTO.mail() != null) this.mail = employeeRequestDTO.mail();
        if(employeeRequestDTO.fonction() != null) this.fonction = employeeRequestDTO.fonction();
    }
}